[#102] map component

SonnyBA opened the pull request:
Fixes: #102 

~TODO:~

- [x]     ~Location permissions~
- [x]     ~Location search~
- [x]     ~Lazy loading~
- [x]     ~Summary patch: https://github.com/open-formulieren/formio-renderer/pull/184#issuecomment-3460943138~
- [x]     ~Translations~
- [x]     ~Documentation~
- [x]     ~Add field to registry~

sergei-maertens commented on src/components/forms/MapField/index.ts:
Do we really need to build/expose this as a low-level field? I think I'd rather to for `src/components/map` as a base and implement the low-level stuff there, and then combine it with the field-stuff in the registry.

There's *a lot* of code here that's directly related to maps, but only tangentially to form fields/Formik. I'd like to separate those more 🤔

SonnyBA commented:
The builds report a decrease in coverage, I don't think we can test more than is already tested though, as the missing coverage seems to be related to using the shapes with multiple points and functions that Leaflet calls:

- [Polyline behavior](https://app.codecov.io/gh/open-formulieren/formio-renderer/pull/184?src=pr&el=tree&filepath=src%2Fcomponents%2Fmap%2FdrawPolylineControl.ts&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=open-formulieren#diff-c3JjL2NvbXBvbmVudHMvbWFwL2RyYXdQb2x5bGluZUNvbnRyb2wudHM=),  these tests were also absent in the SDK and do not seem to be testable
- [Right clicking behavior](https://app.codecov.io/gh/open-formulieren/formio-renderer/pull/184?src=pr&el=tree&filepath=src%2Fcomponents%2Fmap%2FMapField.tsx&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=open-formulieren#diff-c3JjL2NvbXBvbmVudHMvbWFwL01hcEZpZWxkLnRzeA==), this was discussed earlier, these tests were also absent in the SDK and do not seem to be testable, see lines 293-298
- [Location control code](https://app.codecov.io/gh/open-formulieren/formio-renderer/pull/184?src=pr&el=tree&filepath=src%2Fcomponents%2Fmap%2FLeafletMapLocationControl.tsx&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=open-formulieren#diff-c3JjL2NvbXBvbmVudHMvbWFwL0xlYWZsZXRNYXBMb2NhdGlvbkNvbnRyb2wudHN4), these functions are defined so that Leaflet can call these

sergei-maertens commented on src/components/map/NearestAddress.tsx:
```diff
@@ -0,0 +1,58 @@
+  // XXX: this would benefit from caching, but rather than rolling our own we should
+  // probably look into useQuery/tanstack react query
+  const formSettings = useFormSettings();
+  const mapNearestLookup = formSettings?.componentParameters?.map?.mapNearestLookup;
```
can you also prepare a (draft) PR in the SDK that provides/injects this? You can then already validate/test it via `npm run build && npm pack` + installing that tar.gz locally.

sergei-maertens commented on src/registry/map/map.mdx:
````diff
@@ -0,0 +1,72 @@
+```
+import 'MapComponent' from '@open-formulieren/formio-renderer';
+
+const overlayers = [
````
```suggestion
const overlays = [
```

SonnyBA commented:
@sergei-maertens I'll use this open PR for the low-level component. Through 635315f I refactored the component and removed all the form related code and the Formik bindings.

We could move even more code to the registry component and make it similar to how the [`ValueDisplay`](https://github.com/open-formulieren/formio-renderer/pull/184/commits/ecfbe531ec571e431c70b9edcf3768ee0d1c3224#diff-a381581a1c5d3a3e9dc7ac32e52be000f421c305069677d0271156c833444e73L19) was implemented earlier (stripped down to a minimal map), although that will require us to pass through child elements from the registry component (for the location controls for example) to this component.

sergei-maertens reviewed (changes requested):
I think what's still missing here is in `src/index.ts` adding the export for `LeafletMap` so that the SDK can use that too and we can effectively replace most of the map code in the SDK!
